File: bcw4/Basic/Point/Point.cpp
#include <iostream>
#include "Point.h"

Point::Point(double x, double y) : x(0), y(0) {
    this->x = x;
    this->y = y;
}

Point::~Point() {}

double Point::getX() const {
    return x;
}

double Point::getY() const {
    return y;
}

void Point::setX(double value) {
    this->x = value;
}

void Point::setY(double value) {
    this->y = value;
}

double Point::distance(const Point& other) const {
    return hypot(x-other.x, y-other.y);
}

bool Point::operator==(const Point& other) const {
    return (x == other.x && y == other.y);
}

bool Point::operator!=(const Point& other) const {
    return (x != other.x && y != other.y);
}

std::ostream& operator<<(std::ostream& out, const Point& point) {
	out << "(" << point.getX() << ", " << point.getY() << ")";
    return out;
}

File: bcw4/java/Basic/Point/Point.java
public class Point {
    private double x;
    private double y;

    public Point() {
        x = 0;
        y = 0;
    }

    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getX() {
        return x;
    } 

    public double getY() {
        return y;
    } 

    public void setX(double value) {
        x = value;
    }

    public void setY(double value) {
        y = value;
    }

    public double distance(Point other) {
        return Math.hypot(x-other.x, y-other.y);
    }

    public boolean equals(Point other) {
        return x == other.x && y == other.y;
    } 

    public boolean notEquals(Point other) {
        return !equals(other);
    } 

    public String toString() {
        return "(" + x + ", " + y + ")";
    }

    public void print() {
        System.out.format("(%.2f, %.2f)%n", x, y);
    }
}